src/herder/TransactionQueue.cpp

```diff
@@ -218,9 +222,22 @@ isDuplicateTx(TransactionFrameBasePtr oldTx, TransactionFrameBasePtr newTx)
     }
     else if (oldEnv.type() == ENVELOPE_TYPE_TX_FEE_BUMP)
     {
-        auto oldFeeBump =
-            std::static_pointer_cast<FeeBumpTransactionFrame>(oldTx);
-        return oldFeeBump->getInnerFullHash() == newTx->getFullHash();
+        std::shared_ptr<FeeBumpTransactionFrame const> feeBumpPtr{};
+#ifdef BUILD_TESTS
+        if (oldTx->isTestTx())
```

Comment:
Yes, we need the inner hash, and if we call `getFullHash` on the test frame tx it will return the outer hash. We can't a test frame directly to a fee bump, but we have to cast the inner frame pointer, thus the special case.